### ISSUE TITLE
validating webhook fails to install correctly on 1.22+ clusters

### ISSUE THREAD

**Issue by skriss** [kubernetes-sigs/gateway-api#1125]
**What happened**:
I tried to install the validating webhook into a 1.23 cluster, by applying the contents of https://github.com/kubernetes-sigs/gateway-api/tree/v0.4.2/deploy.  The `gateway-api-admission-patch` Job never ran successfully, and the logs for the Job's pod showed:

```
{"err":"the server could not find the requested resource","level":"fatal","msg":"failed getting validating webhook","source":"k8s/k8s.go:48","time":"2022-04-21T21:29:54Z"}
```

**What you expected to happen**:
The webhook to install successfully.

**How to reproduce it (as minimally and precisely as possible)**:
Spin up a 1.23 cluster using your preferred method, then apply the contents of https://github.com/kubernetes-sigs/gateway-api/tree/v0.4.2/deploy.

**Anything else we need to know?**:
The v1beta1 version of the `admissionregistration.k8s.io` API group was dropped in 1.22: https://kubernetes.io/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/#api-changes

But, looks like the kube-webhook-certgen project that's being used to generate certs for the webhook is still using the v1beta1 API: https://github.com/jet/kube-webhook-certgen/blob/master/pkg/k8s/k8s.go#L42-L45

**Comment by robscott** [comment]
Good catch, looks like we need to cherry pick #991 into release-0.4

**Comment by skriss** [comment]
Ah, somehow I overlooked that PR. I can probably just use `main` for now and wait for a 0.5 release with the fix. Thanks for the quick response!

**Comment by skriss** [comment]
Fixed in `main` and `v0.4.3`, closing this out.
